### PR TITLE
Fix MSBuild for Unity 2020

### PR DESCRIPTION
**Pull request by MaxWang-MS**
## Overview

This PR modifies certain MSBuild related scripts so that it works with Unity 2020.

### CODE REVIEW

**Review by davidkline-ms — APPROVED** [review]
this looks good to me

File: Assets/MRTK/Tools/MSBuild/Scripts/TargetFramework.cs [hunk 1]
```diff
@@ -35,7 +35,11 @@ public static string AsMSBuildString(this TargetFramework @this)
                 case TargetFramework.Net20:
                     return "net20";
                 case TargetFramework.Net46:
```
Comment (keveleigh) [inline Assets/MRTK/Tools/MSBuild/Scripts/TargetFramework.cs:37]: Do we need to add a new entry in this enum for `471`?

File: Assets/MRTK/Tools/MSBuild/Scripts/UnityProjectInfo.cs [hunk 2]
```diff
@@ -238,14 +248,31 @@ private CSProjectInfo GetProjectInfo(Dictionary<string, CSProjectInfo> projectsM
...
+                    {
+                        CSProjectInfo newProjInfo = new CSProjectInfo(this, asmDefInfoMap[plugin], projectOutputPath);
+                        if (plugin == plugins[1])
+                        {
+                            newProjInfo.AddDependency(projectsMap[plugins[0]]);
+                        }
+                        projectsMap.Add(plugin, newProjInfo);
+                        toReturn.AddDependency(newProjInfo);
+                    }
+#endif
```
Comment (keveleigh) [inline Assets/MRTK/Tools/MSBuild/Scripts/UnityProjectInfo.cs:275]: What is this block doing?

Comment (keveleigh) [reply Assets/MRTK/Tools/MSBuild/Scripts/UnityProjectInfo.cs:275]: It seems odd that we'd get to this point and not have all the projects we need in the map already

File: Assets/MRTK/Tools/MSBuild/Scripts/TargetFramework.cs (hunk 1 above)
Comment (davidkline-ms) [reply Assets/MRTK/Tools/MSBuild/Scripts/TargetFramework.cs:37]: Good catch. We might.

Comment (MaxWang-MS) [reply Assets/MRTK/Tools/MSBuild/Scripts/TargetFramework.cs:37]: Unity has its own ApiCompatibilityLevel enum and there is not an entry for 471 in Unity 2020. So technically from Unity's standpoint we are still using .net 4.6 and the reason I added 471 here is during msbuild build time the compiler will complain as some of Unity's dependency DLLs is targeting 471.

File: Assets/MRTK/Tools/MSBuild/Scripts/UnityProjectInfo.cs (hunk 2 above)
Comment (MaxWang-MS) [reply Assets/MRTK/Tools/MSBuild/Scripts/UnityProjectInfo.cs:275]: This block creates UGUI csprojs and add them as dependency to the csproj being processed. I agree that it seems odd but with recent major updates of Unity (i.e. 2018 -> 2019 and 2019 -> 2020) UGUI seems to require a little extra handling.

**Review by keveleigh — APPROVED** [review]

**Pull request merged**